### ISSUE TITLE
UI: Only automatically group pods once

### ISSUE THREAD

**Issue by maartengo** [argoproj/argo-cd#15111]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

In an application with more than 15 pods, each change in pod count triggers the following

> Since the number of pods has surpassed the threshold pod count of 15, you will now be switched to the group node view. If you prefer the tree view, you can simply click on the Group Nodes toolbar button to deselect the current view.

![image](https://github.com/argoproj/argo-cd/assets/46345801/0b1886ff-66bd-4e73-9b77-1efce2e3d472)

However, this keeps happening after the first automated grouping. 
![image](https://github.com/argoproj/argo-cd/assets/46345801/1b3cb23f-9460-41d0-a164-c631b4d25779)

**To Reproduce**

1. Deploy 15 pods in a single application. Add sync hooks that deploy additional pods on sync, or use syncwaves to trigger more deployments. We deploy prometheus and thanos in a single application, which has the same effect.
2. Deploy a new version of the pods, or just start a sync without changes
3. You should get multiple 'Since the number of pods...' messages, regardless of the current grouping option

**Expected behavior**

The message should only occur once. 

Another related issue (should I create a new issue for this?) is that it **always** groups the pods whenever the amount of pods change (so long as you have > 15 pods). Manually deselecting the grouping is overwritten.

**Screenshots**

**Version**

```
{
    "Version": "v2.8.0+804d4b8",
    "BuildDate": "2023-08-07T14:25:33Z",
    "GitCommit": "804d4b8ca6bc4c2cf02c5c971aa923ec5b8623f0",
    "GitTreeState": "clean",
    "GoVersion": "go1.20.6",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KustomizeVersion": "v5.1.0 2023-06-19T16:58:18Z",
    "HelmVersion": "v3.12.1+gf32a527",
    "KubectlVersion": "v0.24.2",
    "JsonnetVersion": "v0.20.0"
}
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by maartengo** [comment]
Just found another issue related to this feature: Switching views also triggers the message. E.g. switching from 'list' to 'tree' view also triggers the grouping, _but in reverse?_ 

No matter your original grouping, it will ungroup your pods while still printing the 'Since the number of pods...' message.

**Comment by crenshaw-dev** [comment]
WIP to solve this: https://github.com/argoproj/argo-cd/pull/14821

**Comment by maartengo** [comment]
> WIP to solve this: https://github.com/argoproj/argo-cd/pull/14821

This seems to solve the issue of the repeating notifications. Does this also solve the problem that it keeps trying to group pods, regardless of what the user wants? e.g. if I manually deselect the 'group pods' feature, will it still group pods on the next pod count change?

**Comment by jtcressy** [comment]
Is there any good workaround to disable this "feature"? such as editing a configmap or something to specify some setting? We recently upgraded to 2.8 and this UI change is driving us crazy.

**Comment by crenshaw-dev** [comment]
We got an internal request to improve the experience as well. @ashutosh16 would you like to describe here what we proposed for the internal user?

**Comment by ashutosh16** [comment]
@jtcressy/@maartengo  Based on the feedback, i've  made some changes where the message and group nodes will be displayed once per app and will not be displayed again to the user. Pls see the attach video. It would be great if you can review the changes?

https://github.com/argoproj/argo-cd/assets/11219262/8595a4ae-6372-4f14-b35c-bfa2260309e6

**Comment by maartengo** [comment]
@ashutosh16 this looks great! The only thing that I noticed is that test2 became grouped after you grouped test3. I think that can be a bit confusing, but otherwise its a great improvement.

**Comment by ashutosh16** [comment]
@maartengo

> @ashutosh16 this looks great! The only thing that I noticed is that test2 became grouped after you grouped test3. I think that can be a bit confusing, but otherwise its a great improvement.

That is the default behavior of ArgoCD, If the user selects a grouping node, the user interface will remain in the grouping until you unselect. The `groupNode` feature is a global preference and is applicable to all apps

**Comment by jtcressy** [comment]
@ashutosh16 LGTM! I especially love that you changed it to a tooltip for the groupNodes button, directing the user to the control visually.

**Comment by crenshaw-dev** [comment]
This change is released with 2.8.2. Thanks everyone!
